### DISCUSSION TITLE
every 10th deletion takes longer time

### DISCUSSION THREAD

**Discussion by yulongkelly** [ibis-project/ibis#6374] [Q&A] [answered]
I create a table `ENGINE = MergeTree PRIMARY KEY (date, time) ORDER BY (date, time) SETTINGS disk='s3_cache'` with ~20 columns

I write a python script to delete a specific date and then insert new data.

The delete takes 1s normally but every 10th delete it takes ~15s or even longer

I used lightweight delete (`delete from ...`)

**Comment by lostmygithubaccount** [comment]
hi @yulongkelly, can you provide more information on the backend you're running with and give a reproducible Ibis script where you're observing this behavior?

**Reply by yulongkelly** [reply, marked as answer]
actually I observe this behavior even if I do not use ibis. I think this is a problem on the clickhouse side.

I will close this discussion now.
